.github/workflows/ci.yaml

```diff
@@ -91,7 +91,7 @@ jobs:
           export KAGENT_URL="http://$(kubectl get svc -n kagent kagent-controller -o jsonpath='{.status.loadBalancer.ingress[0].ip}'):8083"
           echo $KAGENT_URL
 
-          go test -v github.com/kagent-dev/kagent/go/test/e2e
+          go test -race -v github.com/kagent-dev/kagent/go/test/e2e
```

Comment:
Will this cause the tests to fail, or do we need to customize the behavior with GORACE variables? What should the behavior be when race detection is enabled?